src/lndk_offers.rs

```diff
@@ -257,6 +269,40 @@ pub async fn connect_to_peer(
     Ok(())
 }
 
+// pay_invoice tries to pay the provided invoice.
+pub(crate) async fn pay_invoice(
+    mut payer: impl InvoicePayer + std::marker::Send + 'static,
+    path: BlindedPath,
+    cltv_expiry_delta: u16,
+    fee_base_msat: u32,
+    payment_hash: [u8; 32],
+    msats: u64,
+) -> Result<(), OfferError<Secp256k1Error>> {
+    let resp = payer
+        .query_routes(path, cltv_expiry_delta, fee_base_msat, msats)
+        .await
+        .map_err(OfferError::RouteFailure)?;
+
+    let send_resp = payer
+        .send_to_route(payment_hash, resp.routes[0].clone())
+        .await
+        .map_err(OfferError::RouteFailure)?;
+
+    if send_resp.status == 1 {
```

Comment:
In a production setting I don't think we'll ever hit an immediate success/failure. Having used LND apis a bunch, I think it's better practise to send the payment in one call and the just immediately track it.